src/framework/global/types/val.cpp

```diff
@@ -56,6 +58,9 @@ Val::Val(const char* str)
 Val::Val(const io::path_t& path)
     : m_val(path.toStdString()), m_type(Type::String) {}
 
+Val::Val(void* ptr)
+    : m_val(ptr::ptr_to_string(ptr)), m_type(Type::Ptr) {}
```

Comment:
What's the reason to store it as a string, instead of as a (64-bit) integer, or as `uintptr_t` after adding that type to the `Value` `std::variant`?

Reply:
First I made void*, then int64 - both are not suitable. The problem is that I need it to convert to QVariant and transfer to QML. But when transferring to QML void* or int64, when receiving back we get the wrong value, the problem is in the js engine, there can be at most 32 bits of value.
Because of this, I decided to transfer through a string, in hex format, to see that it is a pointer. It is possible to make it so that in Val there is void* and in QVariant a string, but I think it is better to have the same.